src/incremental.jl

```diff
@@ -0,0 +1,183 @@
+
+"""
+Init basic C libraries
+"""
+function InitBase()
+  """
+  Base.__init__()
+  """
+end
+
+
+"""
+# Initialize REPL module for Docs
+"""
+function InitREPL()
+  """
+  using REPL
+  Base.REPL_MODULE_REF[] = REPL
+  """
+end
+
+
+
+"""
+Fix for https://github.com/JuliaLang/julia/issues/30479
+"""
+function Fix30479()
+    """
+    _bindir = ccall(:jl_get_julia_bindir, Any, ())::String
+    @eval(Sys, BINDIR = \$(_bindir))
+    @eval(Sys, STDLIB = joinpath(\$_bindir, "..", "share", "julia", "stdlib", string('v', (VERSION.major), '.', VERSION.minor)))
+    """
+end
+
+function Include(path)
+  """
+  M = Module()
+  # Include into anonymous module to not polute namespace
+  @eval(M, (Base.include(\$M, $(repr(path)))))
+  """
+end
+
+"""
+The command to pass to julia --output-o, that runs the julia code in `path` during compilation.
+"""
+function PrecompileCommand(path)
+    InitBase() * Fix30479() * InitREPL() * Include(path)
+end
+
+
+"""
+    extract_flag(flag, jl_cmd = Base.julia_cmd())
+
+Extracts the value for `flag` from the current julia cmd
+"""
+function extract_flag(flag, jl_cmd = Base.julia_cmd())
+    for elem in jl_cmd.exec
+        if startswith(elem, flag)
+            return elem
+        end
+    end
+    return nothing
+end
+
+function extract_flag_value(flag, jl_cmd = Base.julia_cmd())
+    val = extract_flag(flag, jl_cmd)
+    replace(val, flag => "")
+end
+current_systemimage() = extract_flag_value("-J")
+
+"""
+    add_command!(cmd, ignore, jl_cmd, flag, value)
+
+Adds a command to `cmd` - either choosing the julia default for value == nothing,
+or adds value!. If ignore = true, and command not in jl_cmd && value == nothing,
+command gets not added.
+"""
+function add_command!(cmd, ignore, jl_cmd, flag, value)
+    new_cmd = if value == nothing
+        extract_flag(flag, jl_cmd)
+    else
+        short = !startswith(flag, "--")
+        string(flag, short ? "" : "=", value)
+    end
+    if new_cmd == nothing
+        ignore || @warn("Flag $flag not present in julia-cmd, but is set to nothing - can't add the flag")
+    else
+        push!(cmd, new_cmd)
+    end
+end
+
+"""
+    run_julia(
+        code::String;
+        g = nothing, O = nothing, output_o = nothing, J = nothing,
+        startup_file = "no"
+    )
+
+Runs `code` in a new julia command!
+You can overwrite any julia command line flag by setting it to a value.
```

Comment:
Is this really true? Doesn't look like `julia_code_cmd` accepts any kw. Or do you mean just the `g, O` etc flags?

Reply:
Yeah, it's not fully implemented yet ;) Also it should use the cmd struct that stores the actual flags, even if julia wasn't explicitely started with them. Forgot where to find that struct, so I implemented it with julia_cmd for now